test/uk/gov/hmrc/taxhistory/model/nps/HIPIabdSpec.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.taxhistory.model.nps
+
+import org.scalatest.OptionValues
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+import play.api.libs.json.{JsNull, JsObject, JsValue, Json}
+import uk.gov.hmrc.taxhistory.utils.TestUtil
+
+import java.time.LocalDate
+
+class HIPIabdSpec extends TestUtil with AnyWordSpecLike with Matchers with OptionValues {
```

Comment:
can we also add the something like the following?

Required fields are missing
"fail to parse HIPIabd when required field nino is missing" in {
  val invalidJson = Json.parse("""{ "type": "foo (8)" }""")
  assertThrows[Exception](invalidJson.as[HIPIabd])
}

Test unknown payment frequency
"return None for unknown paymentFrequency string" in {
  val json = Json.parse(iabdJsonResponse).as[JsObject] + ("paymentFrequency" -> Json.toJson("UNKNOWN"))
  json.as[HIPIabd].paymentFrequency shouldBe None
}

Round-trip serialization test
"round-trip JSON serialization should yield the same HIPIabd" in {
  val original = Json.parse(iabdJsonResponse).as[HIPIabd]
  val json = Json.toJson(original)
  json.as[HIPIabd] shouldBe original
}